cpp/src/arrow/array-binary-test.cc

```diff
@@ -85,268 +94,132 @@ class TestStringArray : public ::testing::Test {
+    } else if (std::is_same<TypeClass, LargeBinaryType>::value) {
+      ASSERT_EQ(Type::LARGE_BINARY, type->id());
+      ASSERT_EQ(Type::LARGE_BINARY, this->strings_->type_id());
     } else {
-      ASSERT_EQ(expected_[i], strings_->GetString(i));
+      FAIL();
     }
   }
-}
-
-TEST_F(TestStringArray, TestEmptyStringComparison) {
-  offsets_ = {0, 0, 0, 0, 0, 0};
-  offsets_buf_ = Buffer::Wrap(offsets_);
-  length_ = static_cast<int64_t>(offsets_.size() - 1);
-
-  auto strings_a = std::make_shared<StringArray>(length_, offsets_buf_, nullptr,
-                                                 null_bitmap_, null_count_);
-  auto strings_b = std::make_shared<StringArray>(length_, offsets_buf_, nullptr,
-                                                 null_bitmap_, null_count_);
-  ASSERT_TRUE(strings_a->Equals(strings_b));
-}
-
-TEST_F(TestStringArray, CompareNullByteSlots) {
-  StringBuilder builder;
-  StringBuilder builder2;
-  StringBuilder builder3;
-
-  ASSERT_OK(builder.Append("foo"));
-  ASSERT_OK(builder2.Append("foo"));
-  ASSERT_OK(builder3.Append("foo"));
-
-  ASSERT_OK(builder.Append("bar"));
-  ASSERT_OK(builder2.AppendNull());
-
-  // same length, but different
-  ASSERT_OK(builder3.Append("xyz"));
-
-  ASSERT_OK(builder.Append("baz"));
-  ASSERT_OK(builder2.Append("baz"));
-  ASSERT_OK(builder3.Append("baz"));
-
-  std::shared_ptr<Array> array, array2, array3;
-  FinishAndCheckPadding(&builder, &array);
-  ASSERT_OK(builder2.Finish(&array2));
-  ASSERT_OK(builder3.Finish(&array3));
-
-  const auto& a1 = checked_cast<const StringArray&>(*array);
-  const auto& a2 = checked_cast<const StringArray&>(*array2);
-  const auto& a3 = checked_cast<const StringArray&>(*array3);
-
-  // The validity bitmaps are the same, the data is different, but the unequal
-  // portion is masked out
-  StringArray equal_array(3, a1.value_offsets(), a1.value_data(), a2.null_bitmap(), 1);
-  StringArray equal_array2(3, a3.value_offsets(), a3.value_data(), a2.null_bitmap(), 1);
 
-  ASSERT_TRUE(equal_array.Equals(equal_array2));
-  ASSERT_TRUE(a2.RangeEquals(equal_array2, 0, 3, 0));
-
-  ASSERT_TRUE(equal_array.Array::Slice(1)->Equals(equal_array2.Array::Slice(1)));
-  ASSERT_TRUE(
-      equal_array.Array::Slice(1)->RangeEquals(0, 2, 0, equal_array2.Array::Slice(1)));
-}
-
-TEST_F(TestStringArray, TestSliceGetString) {
-  StringBuilder builder;
-
-  ASSERT_OK(builder.Append("a"));
-  ASSERT_OK(builder.Append("b"));
-  ASSERT_OK(builder.Append("c"));
-
-  std::shared_ptr<Array> array;
-  ASSERT_OK(builder.Finish(&array));
-  auto s = array->Slice(1, 10);
-  auto arr = std::dynamic_pointer_cast<StringArray>(s);
-  ASSERT_EQ(arr->GetString(0), "b");
-}
-
-// ----------------------------------------------------------------------
-// String builder tests
-
-class TestStringBuilder : public TestBuilder {
- public:
-  void SetUp() {
-    TestBuilder::SetUp();
-    builder_.reset(new StringBuilder(pool_));
+  void _TestListFunctions() {
```

Comment:
nit: I'm not a huge fan of the _ convention in cpp code, is it necessary for some reason?

Reply:
I wanted to denote that these are not test functions directly executed by gtest. But perhaps it's not necessary. My text editor (Kate) displays C++ methods prefixed with an underscore in a different colour, so perhaps these are reserved in some way?